chia/wallet/util/compressed_types.py

```diff
@@ -0,0 +1,71 @@
+from dataclasses import dataclass
+
+from chia.types.blockchain_format.program import Program
+from chia.types.coin_spend import CoinSpend
+from chia.types.spend_bundle import SpendBundle
+from chia.util.streamable import Streamable, streamable
+from chia.wallet.util.puzzle_compression import PuzzleCompressor
+
+
+@dataclass(frozen=True)
+@streamable
+class CompressedPuzzle(Streamable):
+    compressed_puzzle: Program
+
+    @classmethod
+    def compress(cls, puzzle: Program, compressor=PuzzleCompressor()):
+        return cls(Program.to(compressor.serialize(puzzle)))
+
+    def decompress(self, compressor=PuzzleCompressor()) -> Program:
```

Comment:
Ditto. No default arguments. Default arguments like this are secret globals that will regret later.